agent/consul/rpc.go

```diff
@@ -289,6 +290,23 @@ func (s *Server) forward(method string, info structs.RPCInfo, args interface{},
 	// Handle DC forwarding
 	dc := info.RequestDatacenter()
 	if dc != s.config.Datacenter {
+		// Local tokens only work within the current datacenter. Check to see
+		// if we are attempting to forward one to a remote datacenter and strip
+		// it, falling back on the anonymous token on the other end.
+		if token := info.TokenSecret(); token != "" {
+			done, ident, err := s.ResolveIdentityFromToken(token)
+			if done {
+				if err != nil && !acl.IsErrNotFound(err) {
+					return false, err
+				}
+				if ident != nil && ident.IsLocal() {
+					// Strip it from the request.
+					info.SetTokenSecret("")
+					defer info.SetTokenSecret(token)
```

Comment:
I was wondering why we need to put it back, but then I saw that `RPCInfo` is an interface and interfaces are passed by reference.